centrifuge-app/src/pages/Loan/CorrectionForm.tsx

```diff
@@ -0,0 +1,244 @@
+import { ActiveLoan, CurrencyBalance, Pool, Price } from '@centrifuge/centrifuge-js'
+import { useCentrifugeApi, useCentrifugeTransaction, wrapProxyCallsForAccount } from '@centrifuge/centrifuge-react'
+import { Button, CurrencyInput, Shelf, Stack, Text, TextInput } from '@centrifuge/fabric'
+import BN from 'bn.js'
+import Decimal from 'decimal.js-light'
+import { Field, FieldProps, Form, FormikProvider, useFormik } from 'formik'
+import * as React from 'react'
+import { combineLatest, switchMap } from 'rxjs'
+import { FieldWithErrorMessage } from '../../components/FieldWithErrorMessage'
+import { Dec } from '../../utils/Decimal'
+import { formatBalance } from '../../utils/formatting'
+import { useFocusInvalidInput } from '../../utils/useFocusInvalidInput'
+import { useAvailableFinancing } from '../../utils/useLoans'
+import { useBorrower } from '../../utils/usePermissions'
+import { usePool } from '../../utils/usePools'
+import { combine, max, maxPriceVariance, positiveNumber, required } from '../../utils/validation'
+import { useChargePoolFees } from './ChargeFeesFields'
+import { isCashLoan, isExternalLoan, isInternalLoan } from './utils'
+
+export type CorrectionValues = {
+  principal: number | '' | Decimal
+  price: number | '' | Decimal
+  quantity: number | '' | Decimal
+  fees: { id: string; amount: '' | number | Decimal }[]
+  reason: string
+}
+
+export function CorrectionForm({ loan }: { loan: ActiveLoan }) {
+  const pool = usePool(loan.poolId) as Pool
+  const account = useBorrower(loan.poolId, loan.id)
+  const poolFees = useChargePoolFees(loan.poolId, loan.id)
+  const { initial: availableFinancing } = useAvailableFinancing(loan.poolId, loan.id)
+  const api = useCentrifugeApi()
+  const { execute: doFinanceTransaction, isLoading: isFinanceLoading } = useCentrifugeTransaction(
+    'Adjust asset',
+    (cent) => (args: [poolId: string, loanId: string, values: CorrectionValues], options) => {
+      if (!account) throw new Error('No borrower')
+      const [poolId, loanId, values] = args
+
+      let principal
+      let isIncrease
+      if (isExternalLoan(loan)) {
+        const oldQuantity = loan.pricing.outstandingQuantity
+        const price = CurrencyBalance.fromFloat(values.price, pool.currency.decimals)
+        const quantity = Price.fromFloat(values.quantity)
+        const amount = new CurrencyBalance(price.mul(new BN(quantity.toDecimal().toString())), pool.currency.decimals)
+        isIncrease = oldQuantity.lt(quantity)
+        const diff = amount.sub(oldQuantity)
+        principal = {
+          external: {
+            price: price.toString(),
+            quantity: (isIncrease ? diff : diff.neg()).toString(),
```

Comment:
```suggestion
            quantity: (!diff.isNeg() ? diff : diff.neg()).toString(),
```
Wouldn't this be the same and spare the `isIncrease` variable? Don't think it affect readability

Reply:
yea will get rid of it

Reply:
ah no, the isIncrease variable is needed on line 61